File: Ruby/Weapon.rb
#encoding: UTF-8

require_relative 'Dice'
require_relative 'CombatElement'

module Irrgarten
	class Weapon < CombatElement
		def initialize(power,uses)
			super(power,uses)
		end
		
		public
		def attack
			produce_effect
		end
		
		def to_string
			"W"+super
		end
		
		public_class_method :new
	end
end

File: Java/src/irrgarten/Weapon.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package irrgarten;

/**
 * @author jorge
 */

public class Weapon extends CombatElement{
    // Constructor
    public Weapon (float power, int uses){
        super(power,uses);
    }
    
    
    // Métodos públicos
    public float attack(){
        return produceEffect();
    }
    
    @Override
    public String toString(){
        return "W"+super.toString();
    }
}
